torch/linalg/__init__.py

```diff
@@ -8,6 +8,51 @@
 # Note: This not only adds doc strings for functions in the linalg namespace, but
 # also connects the torch.linalg Python namespace to the torch._C._linalg builtins.
 
+cholesky = _add_docstr(_linalg.linalg_cholesky, r"""
+linalg.cholesky(input) -> Tensor
+
+Returns the Cholesky decomposition.
+
+Computes the Cholesky decomposition of a Hermitian (or symmetric for real-valued matrices)
+positive-definite matrix :math:`A` or for batches of Hermitian positive-definite matrices.
+The returned matrix ``L`` is lower-triangular, and
+the decomposition has the form:
+
+.. math::
+
+    A = LL^H
+
+If :attr:`input` is a batch of Hermitian positive-definite
+matrices, then the returned tensor will be composed of lower-triangular Cholesky factors
+of each of the individual matrices.
+
+.. note:: If the :attr:`input` is not Hermitian positive-definite matrix a RuntimeError is raised
+          saying that the input is singular and mentioning which minor of the input matrix is not positive-definite.
+
+.. note::
+    Supports real and complex inputs.
+    Backpropagation for complex inputs is only supported on the CPU.
+
+Args:
+    input (Tensor): the input tensor :math:`A` of size :math:`(*, n, n)` where `*` is zero or more
```

Comment:
Calling the tensor "A" here isn't helpful and possibly misleading if input is a batch of matrices. Maybe you can say, "a Hermitian positive-definite matrix or batch of such matrices."? 

If multiple matrices are passed as input, does each of them have to have the same size? That requirement (or lack thereof) should be mentioned somewhere in the documents.

Reply:
I'll remove the "A".
The input tensor must have the shape of the form `(*, n, n)`, the last two dimensions are equal, i.e. batch of square matrices. There is a check for that and tests. I could add that matrices should be square, but that's already implicit by calling a matrix positive-definite.

Reply:
And it's not technically possible to pass multiple matrices as the input of different size, as we don't support list of tensors and consider the last two dimensions of a tensor as a matrix we process.

Reply:
Right. Good point. I've been in too many discussions about nested tensors, which can be ragged, recently. My mistake.